Commons/src/main/java/eu/neclab/ngsildbroker/commons/controllers/EntryControllerFunctions.java

```diff
@@ -188,8 +188,9 @@ private static ResponseEntity<String> generateBatchResultReply(BatchResult resul
 		if (result.getSuccess().isEmpty()) {
 			status = HttpStatus.BAD_REQUEST;
 		}
-		if (body == null) {
-			return ResponseEntity.status(status).build();
+		if (result.getFails().isEmpty() && !result.getSuccess().isEmpty()) {
```

Comment:
Only updated should still return an empty body. Add another check if reponsecode is 204 than no body shall be returned.